### DISCUSSION TITLE
Fan Speed

### DISCUSSION THREAD

**Discussion by achecchia** [jason0x43/hacs-hubitat#212] [General]
So with the release of HA that gave us "low medium high" fan speed buttons on the UI, I was wondering if there is something that needs to be done so HA can recognize my fan speed controller in Hubitat to show as such in HA. I have some fans in HE that show in HA as a slider (or dimmer), yet I have fans directly connected via HA that show the buttons. 

Can this be done?

**Comment by simondsmason** [comment]
I think this is related.  I use the Bond hub for my fans.  They show in Hubitat as fan devices with speed control, etc.  When I bring them over to Home Assistant they show only as an on/off switch - no speed control.  Also, when I turn the fans on in Hubitat the on/off switch does not update.
